lms/djangoapps/certificates/tests/test_tasks.py

```diff
@@ -33,3 +36,20 @@ def test_cert_task_missing_args(self, missing_arg):
         with patch('lms.djangoapps.certificates.tasks.User.objects.get'):
             with self.assertRaisesRegexp(KeyError, missing_arg):
                 generate_certificate(**kwargs)
+
+    @patch('lms.djangoapps.certificates.tasks.generate_certificate.retry')
+    @patch('lms.djangoapps.certificates.tasks.generate_user_certificates')
+    def test_cert_task_retry(self, generate_user_certs_mock, mock_retry):
+        course_key = 'course-v1:edX+CS101+2017_T2'
+        student = UserFactory()
+
+        generate_certificate(student=student.id, course_key=course_key, expected_verification_status='approved')
+
+        SoftwareSecurePhotoVerificationFactory.create(user=student, status='approved')
+
+        self.assertTrue(mock_retry.called)
+
+        generate_user_certs_mock.assert_called_with(
```

Comment:
This shouldn't be called if `self.retry()` is called.  Once you add the `else` as mentioned above, this assertion can go away.